File: solutions/1403/main.py
from typing import * 

class Solution:
    def minSubsequence(self, nums: List[int]) -> List[int]:
        nums.sort(reverse=True)
        res = []
        for i, n in enumerate(nums):
            res.append(n) 
            if sum(res) > sum(nums[i+1:]):
                return res 

if __name__ == "__main__": 
    s = Solution()

    nums = [4,3,10,9,8]
    assert s.minSubsequence(nums) == [10,9] 

File: solutions/1403/main.rs
impl Solution {
    pub fn min_subsequence(nums: Vec<i32>) -> Vec<i32> {
        use std::collections::BinaryHeap; 
        let target = nums.iter().sum::<i32>() / 2; 

        let mut heap = nums.into_iter().collect::<BinaryHeap<i32>>(); 

        let mut ans = Vec::new();
        let mut acc = 0; 
        while acc <= target {
            let t = heap.pop().unwrap(); 
            ans.push(t);
            acc += t
        }
        ans 
    }
}